targets/arm/mikroe/sam/src/spi_master/implementations/implementation_1/hal_ll_spi_master.c

```diff
@@ -0,0 +1,661 @@
+** DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT
+** OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE
+** OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+**
+****************************************************************************/
+/*!
+ * @file  hal_ll_spi_master.c
+ * @brief SPI Master HAL LOW LEVEL layer implementation.
+ */
+
+#include "hal_ll_pmc.h"
+#include "hal_ll_spi_master.h"
+#include "hal_ll_spi_master_pin_map.h"
+
+/*!< @brief Local handle list */
+static volatile hal_ll_spi_master_handle_register_t hal_ll_module_state[SPI_MODULE_COUNT] = {(handle_t *)NULL, (handle_t *)NULL, false};
+
+// ------------------------------------------------------------- PRIVATE MACROS
+
+/*!< @brief Helper macro for getting system clock */
+#define hal_ll_get_system_clock (Get_Fosc_kHz() * 1000)
+
+/*!< @brief Helper macro for getting hal_ll_module_state address */
+#define hal_ll_spi_master_get_module_state_address            ((hal_ll_spi_master_handle_register_t *)*handle)
+/*!< @brief Helper macro for getting module specific control register structure base address // first register address */
+#define hal_ll_spi_master_get_handle                          (hal_ll_spi_master_handle_register_t *)hal_ll_spi_master_get_module_state_address->hal_ll_spi_master_handle
+/*!< @brief Helper macro for getting module specific base address directly from HAL layer handle */
+#define hal_ll_spi_master_get_base_from_hal_handle            ((hal_ll_spi_master_hw_specifics_map_t *)((hal_ll_spi_master_handle_register_t *)\
+                                                              (((hal_ll_spi_master_handle_register_t *)(handle))->hal_ll_spi_master_handle))->hal_ll_spi_master_handle)->base
+
+/*!< @brief Macros defining bit location */
+#define HAL_LL_SPI_CSR_CPOL_BIT       0
+#define HAL_LL_SPI_CR_SPIEN_BIT       0
+#define HAL_LL_SPI_MR_MSTR_BIT        0
+#define HAL_LL_SPI_SR_RDRF_BIT        0
+#define HAL_LL_SPI_SR_TDRE_BIT        1
+#define HAL_LL_SPI_CSR_NCPHA_BIT      1
+#define HAL_LL_SPI_CR_SPIDIS_BIT      1
+#define HAL_LL_SPI_MR_PS_BIT          1
+#define HAL_LL_SPI_MR_PCSDEC_BIT      2
+#define HAL_LL_SPI_CR_SWRST_BIT       7
+#define HAL_LL_SPI_CSR_SCBR_START_BIT 8
+#define HAL_LL_PID_SPI1_BIT           10
+#define HAL_LL_PID_SPI0_BIT           21
+
+/*!< @#brief Macros defining register masks */
+#define HAL_LL_CSR_BITS_MASK          0xF0
+#define HAL_LL_SPI_MR_PCS_MASK        0xE0000
+
+/*!< @brief Macros used for calculating baud rate value */
+#define hal_ll_spi_master_get_divider(_fosc,_speed)             (_fosc/_speed)
+#define hal_ll_spi_master_get_actual_speed(_fosc,_divider)    (_fosc/_divider)
+
+/*!< @brief Default SPI Master bit-rate if no speed is set */
+#define HAL_LL_SPI_MASTER_SPEED_100K 100000
+
+// -------------------------------------------------------------- PRIVATE TYPES
+
+/*!< @brief SPI register structure. */
+typedef struct {
+    uint32_t cr;
```

Comment:
Check file for appropriate use of `hal_ll_base_addr_t` and `hal_ll_pin_name_t`

Reply:
Changed it for all source files where needed.